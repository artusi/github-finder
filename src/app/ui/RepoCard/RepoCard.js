import React from "react";
import PropTypes from "prop-types";
import styles from "./RepoCard.module.css";

export function RepoCard({
  id,
  customClassName,
  title,
  description,
  language,
  stars
}) {
  return (
    <div
      id={`repoCard-${id}`}
      className={`${styles.repoCard} ${customClassName}`}
    >
      <div className={styles.header}>
        <p className={styles.title}>{title}</p>
        <span className={styles.description}>{description}</span>
      </div>
      <div className={styles.footer}>
        <ul className={styles.attributes}>
          <li className={styles.attributeItem}>
            <span className={styles.footerTitle}>Language: </span>
            {language}
          </li>
          <li className={styles.attributeItem}>
            <span className={styles.footerTitle}>Stars: </span>
            {stars}
          </li>
        </ul>
      </div>
    </div>
  );
}

RepoCard.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  description: PropTypes.string,
  language: PropTypes.string,
  stars: PropTypes.number,
  customClassName: PropTypes.string
};

RepoCard.defaultProps = {
  customClassName: "",
  description: "",
  language: "",
  stars: 0
};

export default RepoCard;
