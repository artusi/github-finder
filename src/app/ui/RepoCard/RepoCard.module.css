.repoCard {
  display: flex;
  width: 100%;
  height: 100%;
  flex-direction: column;
  color: #00324d;
  padding: 24px;
  border-radius: 15px;
  background-color: #fff;
  box-shadow: 0 1px 3px 0 rgba(51, 51, 51, 0.2);
}
.header {
  height: 70px;
  margin-bottom: 8px;
}
.title {
  font-size: 14px;
  font-weight: bold;
  margin: 0 0 8px;
  padding: 0;
}
.description {
  font-size: 12px;
  color: #00324d;
  /* autoprefixer: off */
  display: -webkit-box;
  overflow: hidden;
  height: 22px * 2;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
}
.attributes {
  padding: 0;
  margin: 0;
  list-style: none;
}
.attributeItem {
  font-size: 12px;
}
.footerTitle {
  font-weight: bold;
}
