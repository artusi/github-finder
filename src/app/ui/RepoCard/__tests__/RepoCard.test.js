import React from "react";
import { RepoCard } from "../RepoCard";

const initialProps = {
  id: "id",
  title: "My awesome repo"
};

describe("<RepoCard /> - render", () => {
  it("renders", () => {
    const wrapper = shallow(<RepoCard {...initialProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("renders with all props > prefix | defaultValue", () => {
    const allProps = {
      ...initialProps,
      description: "Lololo lol lo lo lo lo lo ",
      customClassName: "customClass",
      stars: 10
    };
    const wrapper = shallow(<RepoCard {...allProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
