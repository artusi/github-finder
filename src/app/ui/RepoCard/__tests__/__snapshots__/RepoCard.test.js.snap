// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RepoCard /> - render renders 1`] = `
<div
  className="repoCard "
  id="repoCard-id"
>
  <div
    className="header"
  >
    <p
      className="title"
    >
      My awesome repo
    </p>
    <span
      className="description"
    />
  </div>
  <div
    className="footer"
  >
    <ul
      className="attributes"
    >
      <li
        className="attributeItem"
      >
        <span
          className="footerTitle"
        >
          Language: 
        </span>
      </li>
      <li
        className="attributeItem"
      >
        <span
          className="footerTitle"
        >
          Stars: 
        </span>
        0
      </li>
    </ul>
  </div>
</div>
`;

exports[`<RepoCard /> - render renders with all props > prefix | defaultValue 1`] = `
<div
  className="repoCard customClass"
  id="repoCard-id"
>
  <div
    className="header"
  >
    <p
      className="title"
    >
      My awesome repo
    </p>
    <span
      className="description"
    >
      Lololo lol lo lo lo lo lo 
    </span>
  </div>
  <div
    className="footer"
  >
    <ul
      className="attributes"
    >
      <li
        className="attributeItem"
      >
        <span
          className="footerTitle"
        >
          Language: 
        </span>
      </li>
      <li
        className="attributeItem"
      >
        <span
          className="footerTitle"
        >
          Stars: 
        </span>
        10
      </li>
    </ul>
  </div>
</div>
`;
